### ISSUE TITLE
Refresh Tokens are invalid after reimport of a realm

### ISSUE THREAD

**Issue by buderjoh** [keycloak/keycloak#31536]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

import-export

### Describe the bug

When a access_token/refresh token is generated via Direct Access Grant for example. 

`curl --location --request POST 'https://localhost:13443/auth/realms/myrealm/protocol/openid-connect/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'grant_type=password' \
--data-urlencode 'client_id=myclient' \
--data-urlencode 'client_secret=mysecret' \
--data-urlencode 'username=user' \
--data-urlencode 'password=password' \`

you can use it as usual

`curl --location --request POST 'https://localhost:13443/auth/realms/myrealm/protocol/openid-connect/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'grant_type=refresh_token' \
--data-urlencode 'client_id=myclient' \
--data-urlencode 'username=user' \
--data-urlencode 'client_secret=mysecret' \
--data-urlencode 'password=password' \
--data-urlencode 'scope=offline_access' \
--data-urlencode <refreshtoken>'`

But if you reimport the realm the kid is changed.

![image](https://github.com/user-attachments/assets/7036e413-bdf5-42ec-9eb6-a4076ee6e212)
 
And if a token is refreshed it cant be verified as the old kid of the token cannot be found.

![image](https://github.com/user-attachments/assets/cd8c2b75-9b16-4bed-86a4-fd3116d46d41)

**Verifying code**

https://github.com/keycloak/keycloak/blob/b215e64cf434e5264b9d9a824b129835b74fbad0/services/src/main/java/org/keycloak/crypto/ServerMacSignatureVerifierContext.java#L33

https://github.com/keycloak/keycloak/blob/b215e64cf434e5264b9d9a824b129835b74fbad0/services/src/main/java/org/keycloak/keys/DefaultKeyManager.java#L123

We noticed this behaviour while using offline sessions and refresh tokens

This worked with Keycloak 23.X we believe this change was introduced with https://github.com/keycloak/keycloak/pull/27030/files#diff-fbba94906756df2265b705566108602915da118a3ab552a51a1fbed79fdff9de

It might be solved with the following PR https://github.com/keycloak/keycloak/pull/30692
as it mentions the following: https://github.com/keycloak/keycloak/pull/30692/files#diff-73cf73f0cfedd6352a69121479567ea3a8aa681a8c99b7d50f24e94fbe46f440R103

But we couldnt verify it yet.

### Version

25.0.0

### Regression

- [ ] The issue is a regression

### Expected behavior

Refresh tokens shouldnt be invalid after a reimport of a realm

### Actual behavior

After a realm import a refresh token is invalid.

![image](https://github.com/user-attachments/assets/febc4709-65f6-4a90-be10-8b83204f9117)

### How to Reproduce?

Get Refresh token (see above)
-> reimport realms via operator or on startup
Try to refresh token -> invalid token

**Comment by rmartinc** [comment]
AFAIK the [operator does not allow to import an existing realm](https://www.keycloak.org/operator/realm-import). Are you deleting the realm before and then re-importing it? If I do locally at boot time (kc.sh import), the realm is deleted (message `Realm 'xxx' already exists. Removing it before import`) and then re-imported. And the session will be lost and the token refresh will never work (not only because of the keys problem you are commenting, just because the sessions are lost when removing/stopping the realm).

To understand why the keys are regenerated, does the import file contains the old keys (`org.keycloak.keys.KeyProvider` component of the export with the `secret` info in it)? (If they aren't there they are regenerated at import time.)

**Comment by buderjoh** [comment]
@rmartinc thanks for your comment.

Yes we are removing the realm before.
We found the issue now. The issue was that we still used config from keycloak 23 where the old keys are stored.
The new version ignores those and always regenerates the keys.

![image](https://github.com/user-attachments/assets/0e2db8f4-22f5-4829-b76b-648563278ff1)

We regenerated the whole config with keycloak 25 and now the keys stay.
